x/quarantine/msgs_test.go

```diff
@@ -0,0 +1,1103 @@
+package quarantine_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/provenance-io/provenance/testutil/assertions"
+	"github.com/provenance-io/provenance/x/quarantine"
+	"github.com/provenance-io/provenance/x/quarantine/testutil"
+)
+
+func TestNewMsgOptIn(t *testing.T) {
+	testAddr0 := testutil.MakeTestAddr("nmoi", 0)
+	testAddr1 := testutil.MakeTestAddr("nmoi", 1)
+
+	tests := []struct {
+		name     string
+		toAddr   sdk.AccAddress
+		expected *quarantine.MsgOptIn
+	}{
+		{
+			name:     "addr 0",
+			toAddr:   testAddr0,
+			expected: &quarantine.MsgOptIn{ToAddress: testAddr0.String()},
+		},
+		{
+			name:     "addr 1",
+			toAddr:   testAddr1,
+			expected: &quarantine.MsgOptIn{ToAddress: testAddr1.String()},
+		},
+		{
+			name:     "nil",
+			toAddr:   nil,
+			expected: &quarantine.MsgOptIn{ToAddress: ""},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := quarantine.NewMsgOptIn(tc.toAddr)
+			assert.Equal(t, tc.expected, actual, "NewMsgOptIn")
+		})
+	}
+}
```

Comment:
This isn't a function, it's a test case. And the expected result is that empty message, not nil. In fact, the `NewMsgOptIn` function should never return nil since it's a construct meant to create that message.